tests/kola-ci-self/tests/kola/autopkgtest-soft-reboot

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# Test for soft-reboot functionality similar to autopkgtest-reboot but using systemctl soft-reboot
+# This test validates that the Kola soft-reboot API works correctly
+set -xeuo pipefail
+
+case "${AUTOPKGTEST_REBOOT_MARK:-}" in
+  "") 
+    echo "test beginning"
+    # Check that boot_id stays the same across soft-reboot
+    INITIAL_BOOT_ID=$(cat /proc/sys/kernel/random/boot_id)
+    echo "Initial boot ID: $INITIAL_BOOT_ID" | logger
+    /tmp/autopkgtest-soft-reboot mark1 
+    ;;
+  mark1) 
+    echo "test in mark1"
+    # Verify boot_id is the same (soft-reboot should not change it)
+    CURRENT_BOOT_ID=$(cat /proc/sys/kernel/random/boot_id)
+    echo "Current boot ID after soft-reboot: $CURRENT_BOOT_ID" | logger
```

Comment:
How is this verified though? I think we need to save the boot id to a persistent file like `/var/cache/kola-boot-id` or something in the first boot and then compare it here.